### PR TITLE
Checkbox to select which stations to include in MIG calc

### PR DESCRIPTION
**Pull request by andrea-williams**
https://wps-pr-1309.apps.silver.devops.gov.bc.ca/hfi-calculator

### CODE REVIEW

**Review by Sybrand — APPROVED** [review]
LGTM. Just please check with @tessmgood re. checkbox colours.

File: web/src/features/hfiCalculator/components/DailyViewTable.tsx [hunk 1]
```diff
@@ -157,8 +161,30 @@ const useStyles = makeStyles({
 export const DailyViewTable = (props: Props): JSX.Element => {
   const classes = useStyles()
 
+  const stationCodesList: number[] = []
```
Comment (conbrad) [inline web/src/features/hfiCalculator/components/DailyViewTable.tsx:164]: Can be simplified to:
```
props.dailiesMap.map(daily => daily.code)
```

File: web/src/features/hfiCalculator/components/DailyViewTable.tsx [hunk 2]
```diff
@@ -214,6 +240,7 @@ export const DailyViewTable = (props: Props): JSX.Element => {
           <Table stickyHeader aria-label="daily table view of HFI by planning area">
             <TableHead>
               <TableRow>
+                <TableCell></TableCell>
```
Comment (conbrad) [inline web/src/features/hfiCalculator/components/DailyViewTable.tsx:243]: Is the empty `TableCell` for spacing?

File: web/src/features/hfiCalculator/components/meanIntensity.ts [hunk 3]
```diff
@@ -3,17 +3,20 @@ import { StationDaily } from 'api/hfiCalculatorAPI'
 
 export const calculateMeanIntensityGroup = (
   area: PlanningArea,
-  dailiesMap: Map<number, StationDaily>
+  dailiesMap: Map<number, StationDaily>,
+  selected: number[]
 ): number | undefined => {
   const stationCodesInPlanningArea: number[] = []
```
Comment (conbrad) [inline web/src/features/hfiCalculator/components/meanIntensity.ts:9]: Can be simplified to:
```
Object.entries(area.stations).map(([, station]) => station.code)

**Review by conbrad — APPROVED** [review]
Nice, just a few minor comments

File: web/src/features/hfiCalculator/components/DailyViewTable.tsx (hunk 2 above)
Comment (andrea-williams) [reply web/src/features/hfiCalculator/components/DailyViewTable.tsx:243]: yep. I should add a comment to that effect tho

File: web/src/features/hfiCalculator/components/DailyViewTable.tsx (hunk 1 above)
Comment (andrea-williams) [reply web/src/features/hfiCalculator/components/DailyViewTable.tsx:164]: according to Typescript, there is no `map()` function available on the `Map` prototype. Go figure.

**Review by tessmgood — APPROVED** [review]
Approving as its EOD, with checkbox colour and deselected FIG border colour changes to come. TY!

File: web/src/features/hfiCalculator/components/DailyViewTable.tsx (hunk 1 above)
Comment (conbrad) [reply web/src/features/hfiCalculator/components/DailyViewTable.tsx:164]: Sorry it should be:
```
Object.entries(props.dailiesMap).map(daily => daily.code)
```

**Comment by andrea-williams** [comment]
> Approving as its EOD, with checkbox colour and deselected FIG border colour changes to come. TY!

Checkbox colour has been changed to blue; FIG border colour also changes depending on selected/deselected. Code's been pushed, Openshift just hasn't finished deploying to dev yet so the changes aren't visible in the PR's URL yet

**Review by JulianForeman — APPROVED** [review]

**Comment by conbrad** [comment]
Out of curiosity, 2 questions:

- Does it make sense to have everything deselected by default? This is subjective and maybe just me but it seems like as a Fire Station prep person I'd want all stations as part of prep, and judiciously deselect stations where it makes sense. Acceptance criteria is ambiguous regarding this though.
- Should we have a select/deselect all button in the header? Right now I have to click every single station I'm interested in for prep.

**Comment by andrea-williams** [comment]
> Out of curiosity, 2 questions:
> 
> * Does it make sense to have everything deselected by default? This is subjective and maybe just me but it seems like as a Fire Station prep person I'd want all stations as part of prep, and judiciously deselect stations where it makes sense. Acceptance criteria is ambiguous regarding this though.
> * Should we have a select/deselect all button in the header? Right now I have to click every single station I'm interested in for prep.
> 
> @tessmgood probably knows about the individual workflows better than I though.

Everything should be selected by default, which is what was going on yesterday. Don't know what changed or how, am looking into that now. Maaaayyybe related to the other failure my PR is experiencing re: pulling forecasts from WFWX.

Will be troubleshooting both issues this morning.

File: api/app/wildfire_one/schema_parsers.py [hunk 4]
```diff
@@ -163,61 +163,43 @@ def generate_station_daily(raw_daily,  # pylint: disable=too-many-locals
     isi = raw_daily.get('initialSpreadIndex', None)
     bui = raw_daily.get('buildUpIndex', None)
     ffmc = raw_daily.get('fineFuelMoistureCode', None)
-    fmc = cffdrs.foliar_moisture_content(
-        station.lat, station.long, station.elevation, get_julian_date_now())
+    cc = raw_daily.get('grasslandCuring')
+
+    # set default values in case the calculation fails (likely due to missing data)
+    fmc, sfc, ros, cfb, hfi, lb_ratio, bros, sixty_minute_fire_size, fire_type, intensity_group = None, None, None, None, None, None, None, None, None, None
```
Comment (conbrad) [inline api/app/wildfire_one/schema_parsers.py:169]: Nice, much nicer than setting them everywhere individually!

File: api/app/wildfire_one/schema_parsers.py [hunk 5]
```diff
@@ -163,61 +163,43 @@ def generate_station_daily(raw_daily,  # pylint: disable=too-many-locals
...
-    except cffdrs.CFFDRSException:
-        sfc = None
-        logger.error('Encountered error when calculating surface_fuel_consumption for station %s', station.code)
-
-    cc = raw_daily.get('grasslandCuring')
-    if cc is None:
-        cc = FUEL_TYPE_DEFAULTS[fuel_type]["CC"]
-    try:
+        if cc is None:
+            cc = FUEL_TYPE_DEFAULTS[fuel_type]["CC"]
```
Comment (conbrad) [inline api/app/wildfire_one/schema_parsers.py:176]: Should we just set cc to this initially, so there's no need to set it to `None` in the first place?

File: api/app/wildfire_one/schema_parsers.py [hunk 6]
```diff
@@ -227,24 +209,18 @@ def generate_station_daily(raw_daily,  # pylint: disable=too-many-locals
...
-    except Exception:
-        logger.error('Encountered error calculating fire_type for station %s', station.code)
-        fire_type = None
 
-    if hfi is None:
-        intensity_group = None
-    else:
-        intensity_group = calculate_intensity_group(hfi)
+        if hfi is None:
+            intensity_group = None
```
Comment (conbrad) [inline api/app/wildfire_one/schema_parsers.py:218]: This is already set to `None` in the beginning of the method, no?

File: web/src/features/hfiCalculator/components/DailyViewTable.tsx [hunk 7]
```diff
@@ -221,6 +226,25 @@ export const DailyViewTable = (props: Props): JSX.Element => {
     return 4
   }
 
+  const errorIconTheme = createTheme({
```
Comment (conbrad) [inline web/src/features/hfiCalculator/components/DailyViewTable.tsx:229]: Nice, I think this is the way we're supposed to override CSS in MaterialUI. It's been a struggle with the previous way we were doing it. When it's scoped to a `ThemeProvider` it only affects the children within it, making style changes much easier to reason about.

File: api/app/wildfire_one/schema_parsers.py [hunk 8]
```diff
@@ -3,6 +3,7 @@
 import math
 import logging
 from datetime import datetime, timezone
+from os import stat
```
Comment (conbrad) [inline api/app/wildfire_one/schema_parsers.py:6]: Accidental import?

**Review by conbrad — CHANGES_REQUESTED** [review]
Looking good

File: api/app/wildfire_one/schema_parsers.py (hunk 5 above)
Comment (andrea-williams) [reply api/app/wildfire_one/schema_parsers.py:176]: we don't exactly set `cc` to None initially; we pull the `grasslandCuring` value from `raw_daily` if it's available. We're just using the fuel type defaults if there's no constant value set for the specific station in WFWX

**Review by conbrad — APPROVED** [review]
🚀

**Pull request merged**